homeassistant/components/devolo_home_network/config_flow.py

```diff
@@ -35,7 +37,16 @@ async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str,
 
     device = Device(data[CONF_IP_ADDRESS], zeroconf_instance=zeroconf_instance)
 
+    device.password = data[CONF_PASSWORD]
+
     await device.async_connect(session_instance=async_client)
+
+    # Try a password protected, non-writing device API call that raises, if the password is wrong.
+    # If only the plcnet API is available, we can continue without trying a password as the plcnet
+    # API does not require a password.
+    if device.device:
+        await device.device.async_uptime()
```

Comment:
this I don't understand. You  are not `try`ing or returning anything. How do you deal with the outcome of this test?

Reply:
Not here, but in line 81 and line 127 I'm catching possible DevicePasswordProtected exceptions. Would it be more understandable, if I catch the exception here and reraise?

Reply:
maybe it's just the comment that's misleading as this is directly over this code. The other thing I don't understand is why are we not checking the updated credentials in reauth_confirm, otherwise we could've maybe re-used that step?

Reply:
That's a very valid point. I will add the verification of the updated credentials.